packages/markers/marker.go

```diff
@@ -373,6 +374,15 @@ func (m *Markers) String() (humanReadableMarkers string) {
 	return structBuilder.String()
 }
 
+// SequenceToString returns a string in the form sequenceID:index;.
+func (m *Markers) SequenceToString() (s string) {
+	m.ForEach(func(sequenceID SequenceID, index Index) bool {
+		s += fmt.Sprintf("%d:%d;", sequenceID, index)
+		return true
+	})
```

Comment:
Concatenating strings like that is very inefficient operation, consider building a slice first and then joining it into the result string.
```suggestion
      parts := make([]string, 0, m.Size())
	m.ForEach(func(sequenceID SequenceID, index Index) bool {
		parts = append(parts, fmt.Sprintf("%d:%d", sequenceID, index))
		return true
	})
	s = strings.Join(parts, ";")
```